File: 200-299/215-kth-largest-element-in-an-array.go
package main

func findKthLargest(nums []int, k int) int {
	// 桶
	var (
		buckets = make([]int, 20001)
		count   int
	)
	for _, num := range nums {
		buckets[num+10000]++
	}

	for i := len(buckets) - 1; i >= 0; i-- {
		count += buckets[i]
		if count >= k {
			return i - 10000
		}
	}
	return 0
}

File: 200-299/215-kth-largest-element-in-an-array.py
# coding: utf-8
from collections import defaultdict


class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        bucket = defaultdict(int)
        for num in nums:
            bucket[num] += 1

        i = 10000
        while i >= -10000:
            if bucket[i] >= k:
                return i
            k -= bucket[i]
            i -= 1
        return 0
